### PR TITLE
meerk40t: init at 0.8.0031

### PR DESCRIPTION
**Pull request by mweinelt**
> MeerK40t (pronounced MeerKat) is a built-from-the-ground-up MIT licensed open-source laser cutting software. It's a replacement for LaserDrw, Corel Laser, and K40 Whisperer.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fooker** [comment]
Shouldn't the update for `ezdxf` go into a separate PR?

**Comment by mweinelt** [comment]
Meerk40t required a newer version, where `exzdf.colors` is available.

**Comment by fleaz** [comment]
meerk40t builds and starts fine, but throws quite a lot of errors when trying to use it

Pushing "Open" in the top left corner crashes the application immediately with `(meerk40t:649701): GLib-GIO-ERROR **: 14:07:10.216: No GSettings schemas are installed on the system`

Pushing Configuration->Config creates the following error:
2022-09-12T14:09:16,206098615+02:00

And moving the GUI around often results in this error:
2022-09-12T14:09:32,059589351+02:00


But I haven't used meerk40t on any other system/OS so far, so I can't say if any of these things are related to NixOS

**Comment by mweinelt** [comment]
Looks like it is unhappy with wxWidgets 4.1, I can try downgrading to 4.0 in a moment. Meerk40t says it wants `wxWidgets>=4.0`, so if this works this is a bug we should report upstream.

https://github.com/meerk40t/meerk40t/blob/0.7.7001/setup.py#L7

**Comment by mweinelt** [comment]
With wxWidgets 4.0 I'm hitting a SIGTRAP in glib.

**Comment by mweinelt** [comment]
Fixed. It needed some glue for running GTK apps.

**Comment by mweinelt** [comment]
Includes #191729 and should wait for its merge.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -11780,6 +11780,13 @@ in {
     };
   };
 
+  wxPython_4_2 = callPackage ../development/python-modules/wxPython/4.2.nix {
+    wxGTK = pkgs.wxGTK32.override {
+      withWebKit = true;
+    };
+  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:11787]: Do we really must do this?

File: pkgs/development/python-modules/wxPython/4.2.nix [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    pillow
+    six
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+
+    export DOXYGEN=${doxygen}/bin/doxygen
+    export PATH="${wxGTK}/bin:$PATH"
+    export SDL_CONFIG="${SDL.dev}/bin/sdl-config"
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/wxPython/4.2.nix:99]: At least doxygen should not be cross friendly here.

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/top-level/python-packages.nix:11906]: No idea.

File: pkgs/development/python-modules/wxPython/4.2.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/python-modules/wxPython/4.2.nix:99]: Instructions unclear.

**Comment by mweinelt** [comment]
This is ready and I'd appreciate a final review by @fleaz and/or @fooker, preferably by checking if interaction with a K40 device actually works.

File: pkgs/development/python-modules/wxPython/4.2.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/wxPython/4.2.nix:101]: doxygen should be in nativeBuildInputs instead.

**Comment by fleaz** [comment]
> This is ready and I'd appreciate a final review by @fleaz and/or @fooker, preferably by checking if interaction with a K40 device actually works.

Building/starting/clicking around worked for me (all my previous mentioned bugs are gone) but I'm out of town for 2 weeks so I sadly can't test it with a real K40 until I'm back :(

**Comment by fleaz** [comment]
FYI:
I tested this PR prior to merging with a real K40 and most of the features appeared to be working :)

**Pull request merged**